### PR TITLE
 Add implementation for getting bitmoji snapcode

### PR DESCRIPTION
**Pull request by doinggodswork**

### CODE REVIEW

File: Live-Snap/Live-Snap/Controllers/FetchSnapcodeViewController.swift [hunk 1]
```diff
@@ -90,18 +90,75 @@ class FetchSnapcodeViewController: UIViewController, UITextFieldDelegate {
...
-                System.shared.snapcode = snapcodeImage
-                let selectPhotoViewController = SelectPhotoViewController()
-                System.shared.appDelegate().pageViewController?.setViewControllers([selectPhotoViewController], direction: .forward, animated: true, completion: nil)
-            } else {
-                // display failure case here
+        // Check if username is valid
+        IsValidSnapchatUsernameEndpointRequest(snapchatUsername: username).start { (isValidUsername: Bool) in
+            guard isValidUsername else {
+                DispatchQueue.main.async {
+                    self.showInvalidSnapchatUsernameError()
```
Comment (OlegAba) [inline Live-Snap/Live-Snap/Controllers/FetchSnapcodeViewController.swift]: Why is this in a dispatchQueue? Are you planning on adding UI updates if an error occurs?

File: Live-Snap/Live-Snap/Networking/IsValidSnapchatUsernameEndpointRequest.swift [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+//
+//  Created by Baby on 3/13/18.
+//  Copyright © 2018 Oleg Abalonski. All rights reserved.
+//
+
+import UIKit
+import Alamofire
+
+class IsValidSnapchatUsernameEndpointRequest {
+
```
Comment (OlegAba) [inline Live-Snap/Live-Snap/Networking/IsValidSnapchatUsernameEndpointRequest.swift]: Unfortunately this approach won't work to check if a snapchat username is valid. We can still use this class to check if a bitmoji is available for the username. Possibly display a toggle box if the bitmoji is available?

File: Live-Snap/Live-Snap/Networking/IsValidSnapchatUsernameEndpointRequest.swift [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+        let url = "https://feelinsonice-hrd.appspot.com/web/deeplink/snapcode?username=\(snapchatUsername)&type=SVG"
+        
+        Alamofire.request(url).validate().responseString { (response: DataResponse<String>) in
+            
+            
+            switch response.result {
+            case .success:
+                
+                if let responseString = response.result.value {
+                    completion(responseString.contains(substring: "clip-path=\"url(#clip)\""))
```
Comment (OlegAba) [inline Live-Snap/Live-Snap/Networking/IsValidSnapchatUsernameEndpointRequest.swift]: Should we implement a check for basic username guidelines? According to Snapchats Law Enforcement Guide,

Snapchat usernames:
● Must be 3-15 characters long
● Can’t contain spaces
● Must begin with a letter
● Can only contain letters, numbers, and the special characters
hyphen ( - ), underscore ( _ ), and period ( . ), EXCEPT that the username:
    ○ Can’t begin with a number, hyphen, underscore, or period
    ○ Can’t end with a hyphen, underscore, or period
● Can’t contain emojis or other symbols such as @, $, #, etc.
● Will appear only in lower-case letters within the app

-https://storage.googleapis.com/snap-inc/privacy/lawenforcement.pdf

File: Live-Snap/Live-Snap/Controllers/FetchSnapcodeViewController.swift [hunk 4]
```diff
@@ -90,18 +90,75 @@ class FetchSnapcodeViewController: UIViewController, UITextFieldDelegate {
...
+            self.getSnapcodeWithBitmojiImage(username: username) { (bitmojiSnapcodeImage: UIImage?) in
+                
+                // Get regular snapcode with ghose in the middle
+                let snapcodeEndpointRequest = SnapcodeEndpointRequest(snapchatUsername: username)
+                snapcodeEndpointRequest.start { (snapcodeImage: UIImage?) in
+                    
+                    if let bitmojiSnapcodeImage = bitmojiSnapcodeImage {
+                        System.shared.snapcode = bitmojiSnapcodeImage
+                    } else if let snapcodeImage = snapcodeImage {
+                        System.shared.snapcode = snapcodeImage
```
Comment (OlegAba) [inline Live-Snap/Live-Snap/Controllers/FetchSnapcodeViewController.swift]: Note that we can get a 800x800 snapcode image using a PNG link and only a 300x300 snapcode image using SVG link. If the bitmoji is not available, we might want to use the PNG image for better quality.

File: Live-Snap/Live-Snap/Controllers/FetchSnapcodeViewController.swift (hunk 1 above)
Comment (doinggodswork) [reply Live-Snap/Live-Snap/Controllers/FetchSnapcodeViewController.swift]: yes

File: Live-Snap/Live-Snap/Controllers/FetchSnapcodeViewController.swift (hunk 4 above)
Comment (doinggodswork) [reply Live-Snap/Live-Snap/Controllers/FetchSnapcodeViewController.swift]: if bitmoji is not available it won't be set here

**Pull request merged**